spec/http_exception_spec.rb

```diff
@@ -47,30 +47,54 @@ class TestException < RuntimeError
   end
 
   describe ".check_response!" do
-    it "raises exception on non-200 response" do
-      expect do
-        described_class.check_response!(invalid_response)
-      end.to raise_error(Http::Exceptions::HttpException)
+    shared_examples 'invalid response' do |error_status_code|
+      context "when the response code is not a 200" do
+        let(:code) { error_status_code }
+
+        it "raises exception" do
+          expect do
+            described_class.check_response!(response)
+          end.to raise_error(Http::Exceptions::HttpException)
+        end
+      end
+
+      it "the raised exception contains the response" do
+        begin
+          described_class.check_response!(response)
+        rescue Http::Exceptions::HttpException => e
+          expect(e.response).to eq(response)
+        end
+      end
     end
 
-    it "the raised exception contains the response" do
-      begin
-        described_class.check_response!(invalid_response)
-      rescue Http::Exceptions::HttpException => e
-        expect(e.response).to eq(invalid_response)
+    shared_examples 'valid response' do |successful_status_code|
+      context "when the response code is a 200" do
+        let(:code) { successful_status_code }
+
+        it "returns the response" do
+          expect(described_class.check_response!(response)).to eq(response)
+        end
       end
     end
 
-    it "returns the response on valid response" do
-      expect(described_class.check_response!(valid_response)).to eq(valid_response)
+    context "when the response code is an integer" do
+      it_behaves_like 'invalid response', 400
+      it_behaves_like 'valid response', 200
+    end
+
+    context "when the response code is a string" do
```

Comment:
@ukd1 I added this context that show's it behaves as expected when the status code is a string
and the one above it to show the old behavior. Is this sufficient?